### PR TITLE
Decorator Tests

### PR DESCRIPTION
**Pull request by nickbar01234**
# Description

Setup configurations to write unit testings with `Vitest`. In addition, I wrote some sample tests using mocks for `withSession` and `withRole` decorators.

## Test

1. Create a `.env.test` file in the root directory.
2. Copy all the environment variables from `.env` and add to `.env.test`. Each environment variable should be prepended with `VITE_`.
3. From the terminal, run `npm run test`

## Possible Downsides

I had to make an additional `tsconfig.build.json` to exclude test files from production build. If we exclude test files in the original `tsconfig.json`, then we don't have the helpful type hints from vscode ts-server.

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -26,4 +26,7 @@ export default defineNextConfig({
       { protocol: "https", hostname: "**.googleusercontent.com" },
     ],
   },
+  typescript: {
+    tsconfigPath: "tsconfig.build.json",
+  },
```
Comment (nickbar01234) [inline next.config.mjs:31]: Specify that build should be done using `tsconfig.build.json`.

File: src/env/server.mjs [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
 /**
  * This file is included in `/next.config.mjs` which ensures the app isn't built with invalid env vars.
  * It has to be a `.mjs`-file to be imported there.
  */
 import { serverSchema } from "./schema.mjs";
 import { env as clientEnv, formatErrors } from "./client.mjs";
 
-const _serverEnv = serverSchema.safeParse(process.env);
+if (process.env.NODE_ENV === "test") {
+  // @ts-expect-error Disable env doesn't exist error
```
Comment (nickbar01234) [inline src/env/server.mjs:9]: Probably should revisit and add `env` to a type declration file

File: tsconfig.build.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "exclude": ["node_modules", "./src/**/*.spec.ts"]
```
Comment (nickbar01234) [inline tsconfig.build.json:3]: Ignore all files that end in `spec.ts`.

File: vitest.config.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import path from "path";
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+  resolve: {
+    alias: [
+      {
+        find: "@api",
```
Comment (nickbar01234) [inline vitest.config.ts:8]: Resolve path aliases that we defined in `tsconfig.json`.

**Review by sburchfield33 — APPROVED** [review]

**Pull request merged**